<!DOCTYPE html>
<html lang="en-us">

<head>
  <meta charset="UTF-8">
  <title>Bioinformatics by tethig</title>
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
  <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
  <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
  <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
</head>

<body>
  <section class="page-header">
    <h1 class="project-name">Bioinformatics</h1>
    <h2 class="project-tagline">Dr Dickins Bioinformatics Teaching</h2>
    <a href="https://github.com/tethig/bioinformatics" class="btn">View on GitHub</a>
    <a href="https://github.com/tethig/bioinformatics/zipball/master" class="btn">Download .zip</a>
    <a href="https://github.com/tethig/bioinformatics/tarball/master" class="btn">Download .tar.gz</a>
  </section>

  <section class="main-content">
    <h1>
<a id="bioinformatics-workshops" class="anchor" href="#bioinformatics-workshops" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Bioinformatics Workshops</h1>

    <p>Welcome to Dr Dickins's Bioinformatics Class</p>

    <h2>
<a id="pre-requisites" class="anchor" href="#pre-requisites" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Pre-requisites</h2>

    <p>It is useful to have a professional text editor. Unfortunately, Notepad (on Windows) or TextEdit (on MacOS) tend to correct spelling. This is <strong>not</strong> something we want when handling sequences. In addition, Notepad has a problem displaying
      newline characters <a href="https://stackoverflow.com/a/1552775">which tend to differ between platforms</a>. This is a pain when viewing FASTA files.</p>

    <p>If you are using an NTU PC, I recommend launching an instance of "Notepad++" via NTU's Software Centre (icon on Desktop). If you are using your own computer running Windows you can get it from <a href="https://notepad-plus-plus.org">this link</a>.
      For those of you with a technical bent, and on any OS, try GitHub's own <a href="https://atom.io">Atom editor</a>.</p>

    <h2>
<a id="links-to-content" class="anchor" href="#links-to-content" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Links to Content</h2>

    <p>Please follow the links below to the appropriate class:</p>

    <table>
      <thead>
        <tr>
          <th align="center">Workshop #</th>
          <th align="center">Workshop Title &amp; Link</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td align="center">1</td>
          <td align="center">BLAST Searching (on NOW only)</td>
        </tr>
        <tr>
          <td align="center">2</td>
          <td align="center"><a href="aligntree.html">Sequence Retrieval &amp; Alignment</a></td>
        </tr>
        <tr>
          <td align="center">3</td>
          <td align="center"><a href="genbrowse.html">Genome Browsers</a></td>
        </tr>
        <tr>
          <td align="center">Help!</td>
          <td align="center"><a href="repnote.html">Getting Repeat Data from UCSC</a></td>
        </tr>
      </tbody>
    </table>

    <h2>
    <a id="improvements" class="anchor" href="#improvements" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Improvements</h2>

        <h3>Literature/Guides</h3>
        <p>I have identified <a href="http://dx.doi.org/10.1016/S0168-9525(03)00112-4">this very useful guide</a> (available from NTU library). This runs through phylogenetic tree construction methods and highlights some issues that are currently under-emphasised in the workshop. These include the ability to rotate around nodes, placement of the root (think about your most distant sequence), the usefulness of bootstrap measures and the common, but important, issue of long-branch attraction (consider your interpretation).</p>
        <p>Note that the <a href="https://www.ebi.ac.uk/Tools/phylogeny/simple_phylogeny/">Simple Phylogeny tool</a> used does not appear to offer bootstrap options and you may wish to try the tools/workflows on <a href="http://www.phylogeny.fr">phylogeny.fr</a> for this purpose.</p>
        <p>Recall also the reference given in my Paralogy lecture:<br><code>Koonin, E. V. (2005) Orthologs, paralogs, and evolutionary genomics. Annu. Rev. Genet., 39: 309-338.</code></p>

          <h3>Alternative Software</h3>
          <p>The following software could be tried:
            <ul>
              <li><a href="https://mafft.cbrc.jp/alignment/server/">MAFFT (for alignment)</a>: click on "Algorithms" for an explanation of its approach.</li>
              <li><a href="http://molevol.cmima.csic.es/castresana/Gblocks_server.html">Gblocks server</a>: if you wish to avoid phylogeny.fr.</li>
            </ul>
          Many excellent GUIs (e.g., MEGA) and command-line tools (e.g., MrBayes, HyPhy) exist offering more sophisticated functionality. In addition there are specialist tools for large-scale phylogenies (e.g., Parsnp, RAxML).</p>

    <footer class="site-footer">
      <span class="site-footer-owner"><a href="https://github.com/tethig/bioinformatics">Bioinformatics</a> is maintained by <a href="https://github.com/tethig">tethig</a>.</span>

      <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>
    </footer>

  </section>


</body>

</html>
